src/main/java/com/commercetools/sync/customers/utils/CustomerUpdateActionUtils.java

```diff
@@ -349,17 +375,30 @@ private CustomerUpdateActionUtils() {
 
         return buildSetStoreUpdateAction(oldStores, newStores)
             .map(Collections::singletonList)
-            .orElseGet(() -> {
-                if (oldStores != null && newStores != null) {
-                    final List<UpdateAction<Customer>> updateActions =
-                        buildRemoveStoreUpdateActions(oldStores, newStores);
+            .orElseGet(() -> prepareStoreActions(oldStores, newStores));
+    }
+
+    private static List<UpdateAction<Customer>> prepareStoreActions(
```

Comment:
Why do you have both setStores and add/remove stores? Isn't it enough to have setStores?

Reply:
It's enough, `setStores` can do add and remove together, but here I want to expose both and also assuming it's better for API's performance. I was thinking those are public methods but realized it's not :) https://github.com/commercetools/commercetools-sync-java/pull/601/commits/49848f378825d7331a484f5f96e7e753d38d0af2

Reply:
Then I guess you can also update this part
```
        return buildSetStoreUpdateAction(oldStores, newStores)
            .map(Collections::singletonList)	
            .orElseGet(() -> prepareStoreActions(oldStores, newStores));
```

Reply:
Let me check what could be done in here.

Reply:
I think you can remove this part:
```
            .orElseGet(() -> prepareStoreActions(oldStores, newStores));
```

Reply:
Anyways I will keep this as is, setStores will not have a separate public method, 
so `buildStoreUpdateActions` will prepare all actions, which is acceptable for the purpose of the `setStores`. 
`buildRemoveStoreUpdateActions` will prepare only  `removeStore` and same for 
`buildAddStoreUpdateActions` will prepare only `addStore` actions.

Reply:
I added a comment, maybe it's more clear now: https://github.com/commercetools/commercetools-sync-java/pull/601/files#r505397040